### PR TITLE
[JENKINS-36193] Raise exception when trying to build a job that is disabled

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This PR is built on the top of https://github.com/jenkinsci/jenkins/pull/2432. It just adds new tests, so additional full review is not required

[JENKINS-36193](https://issues.jenkins-ci.org/browse/JENKINS-36193)

CC @godfath3r

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
CC @jtnord , who may +1 tests

File: test/src/test/java/jenkins/model/ParameterizedJobMixInTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    
+    @Test
+    public void doBuild_shouldFailWhenInvokingDisabledProject() throws Exception {
+        final FreeStyleProject project = j.createFreeStyleProject();
+        project.doDisable();
+        
+        final JenkinsRule.WebClient webClient = j.createWebClient();
+        webClient.assertFails(project.getUrl() + "build", HttpServletResponse.SC_CONFLICT);
```
Comment (jtnord) [inline test/src/test/java/jenkins/model/ParameterizedJobMixInTest.java]: looks good - but probably should assert the queue is empty, there are no builds in progress and there are no builds for the project - after all you can return whatever code you like and you could still perform the build....

File: test/src/test/java/jenkins/model/ParameterizedJobMixInTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    
+    @Test
+    @Issue("JENKINS-36193")
+    public void doBuildWithParameters_shouldFailWhenInvokingDisabledProject() throws Exception {
+        final FreeStyleProject project = j.createFreeStyleProject();
+        project.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("FOO", "BAR")));
+        project.doDisable();
+        
+        final JenkinsRule.WebClient webClient = j.createWebClient();
+        webClient.assertFails(project.getUrl() + "buildWithParameters", HttpServletResponse.SC_CONFLICT);
```
Comment (jtnord) [inline test/src/test/java/jenkins/model/ParameterizedJobMixInTest.java]: looks good - but probably should assert the queue is empty, there are no builds in progress and there are no builds for the project - after all you can return whatever code you like and you could still perform the build....

**Comment by jtnord** [comment]
:+1: for having a test.

**Comment by oleg-nenashev** [comment]
Build timed out :(

**Comment by oleg-nenashev** [comment]
@jtnord I'll follow up later with test improvements for this class (it needs more love in any case)

**Pull request merged**